src/IceRpc.Protobuf/InvokerExtensions.cs

```diff
@@ -45,21 +42,21 @@ public static class InvokerExtensions
         this IInvoker invoker,
         ServiceAddress serviceAddress,
         string operation,
-        PipeReader payload,
-        PipeReader? payloadContinuation,
-        MessageParser<T> responseMessageParser,
+        IMessage inputMessage,
+        ProtobufEncodeOptions? encodeOptions,
+        ResponseDecodeFunc<T> responseDecodeFunc,
```

Comment:
It's not clear to me why we need this ResponseDecodeFunc. Why not 4 InvokeAsync overloads, for all combinations of unary / stream request / response?

Reply:
The issue with using overloads for the `InvokeAsync` method is the ambiguity introduced when overloads differ only in their return types. Specifically:

- `InvokeAsync` for unary RPC:
  * Input: `IMessage<T>, MessageParser<T>`
  * Return: `IMessage<T>`
  
- `InvokeAsync` for client-streaming RPC:
  * Input: `IAsyncEnumerable<IMessage<T>>`
  * Return: `IMessage<T>`
  
- `InvokeAsync` for server-streaming RPC:
  * Input: `IMessage<T>`
  * Return: `IAsyncEnumerable<IMessage<T>>`
  
- `InvokeAsync` for bidirectional-streaming RPC:
  * Input: `IAsyncEnumerable<IMessage<T>>`
  * Return: `IAsyncEnumerable<IMessage<T>>`

There's ambiguity between the overloads for unary and server-streaming RPCs. Similarly, ambiguity exists between the overloads for client-streaming and bidirectional-streaming RPCs.

We can of course give these methods different names:

- `UnaryInvokeAsync`
- `ClientStreamingInvokeAsync`
- `ServerStreamingInvokeAsync`
- `BidirectionalStreamingInvokeAsync`

Which I think would be fine, the principal caller of these methods is the Protobuf client-generated code.

Reply:
Fixed the code to use separate extension methods as mentioned above.

Reply:
I would prefer: InvokeXxxAsync
with Xxx = Unary, ClientStreaming, ServerStream, BidiStreaming

per the names in https://protobuf.com/docs/language-spec#methods